### ISSUE TITLE
False positive RedundantCondition in php 7 for return type of trim and others

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2143]
Observed: the below example emits TypeDoesNotContainType and RedundantCondition
Expected: Should not warn, unless the php version is 8.0 or newer (i.e. Reflection return type exists)

```php
<?php
function accept_arg($arg) {
    $v = trim($arg);
    if (is_string($v)) {
        echo "Saw $v\n";
    } else {
        echo "Invalid arg type: " . gettype($arg) . "\n";
    }
}
// Invalid arg type: array
accept_arg([]);
```

```
php > echo (new ReflectionFunction('trim'));
Function [ <internal:standard> function trim ] {

  - Parameters [2] {
    Parameter #0 [ <required> $str ]
    Parameter #1 [ <optional> $character_mask ]
  }
  - Return [ string ]
}
php > echo PHP_VERSION;
8.0.0-dev
```

**Comment by muglug** [comment]
It's slightly buggy, but 99.9% of the time a `mixed` type passed to `trim` will result in a string.
